src/modules/org.junit.jupiter.engine/module-info.java

```diff
@@ -0,0 +1,17 @@
+/*
+ * Copyright 2015-2019 the original author or authors.
+ *
+ * All rights reserved. This program and the accompanying materials are
+ * made available under the terms of the Eclipse Public License v2.0 which
+ * accompanies this distribution and is available at
+ *
+ * https://www.eclipse.org/legal/epl-v20.html
+ */
+
+module org.junit.jupiter.engine {
+	requires transitive org.junit.jupiter.api;
+	requires transitive org.junit.platform.engine;
+
+	provides org.junit.platform.engine.TestEngine
+			with org.junit.jupiter.engine.JupiterTestEngine;
```

Comment:
Sad, but true.

Mh, can we move the `JupiterTestEngine` into a different (non-exported) package or remove the `public` access modifier?

Reply:
We cannot remove the `public` access modifier for `JupiterTestEngine` since the `ServiceLoader` (on Java 8) will no longer be able to load it.

It's only possible to make a _service_ `private` starting with Java 9.

Reply:
_Let's move to 9+ then..._

So, moving the engine implementation would work.

Reply:
> _Let's move to 9+ then..._

Java 11 baseline for JUnit 6... at some point in the future.

> So, moving the engine implementation would work.

Yes, that should work.